src/Build/Graph/ProjectGraph.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Build.Exceptions;
+
+namespace Microsoft.Build.Graph
+{
+    /// <summary>
+    /// Represents a graph of evaluated projects.
+    /// </summary>
+    internal class ProjectGraph
```

Comment:
The class is internal for now to avoid anything leaking into the public API prematurely. It wouldn't be the end of the world if it did, but it might be easier to make breaking changes early if it's all internal.

Reply:
Sounds good as long as we don't have to do too many additional `InternalsVisibleTo` tricks.

Reply:
Agreed. I think once we start needing things to be public (without adding shenanigans / hacks), we'll make them public. I anticipate at least a few changes before that happens though :)